### PR TITLE
[WIP] Migrate to WebComponents

### PR DESCRIPTION
**Pull request by mikeal**
Opening this early as I start the migration. It's a big code re-org so I want people to be aware of it.

I may also move to webpack in the same branch because the rebuild times are killing me.

### CODE REVIEW

File: lib/components.js [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+    this.storage = new UserStorage()
+    this.masterSoundOutput = waudio(true)
+
+    this.mainContainer = bel`
+    <div id="main-container" class="ui main text container">
+      <div id="messages-container"></div>
+      <div id="tracks-container" class="ui items"></div>
+      <div id="audio-container" class="ui special cards"></div>
+    </div>`
+    this.appendChild(this.mainContainer)
```
Comment (bmathews) [inline lib/components.js:58]: Adding children inside of the constructor of a custom element isn't valid.

You'll get an error like this: http://stackoverflow.com/questions/40181683/failed-to-execute-createelement-on-document-the-result-must-not-have-childr

If created via `document.createElement('roll-call')`.

Comment (bmathews) [reply lib/components.js:58]: Here's an example: https://jsfiddle.net/g3nc786z/

**Comment by MarcL** [comment]
This is a good idea. Finding your way round the current code is a bit tricky.
Will pull this when you've merged and start adding some of the additional functionality.
Also happy to start adding tests too.

**Comment by mikeal** [comment]
Closing this.

While I like this style of component development it's a little too early to move such a large code base to it. I'll be building out some other stuff with it in the meantime and if all goes well I will come back and migrate roll call.

**Pull request closed without merging**